File: 20190917/20190923/665. 非递减数列.py
'''
Description: 非递减数列
Author: AskeyNil
CreateDate: 2019-09-23 15:24:02
LastEditors: AskeyNil

*********************************
**                             **
**     　  你只有足够努力     　  **
**       才能看上去毫不费力       **
**                             **
*********************************

'''


# 特判
class Solution:
    def checkPossibility(self, nums: [int]) -> bool:
        p = None
        for index in range(len(nums) - 1):
            if nums[index] > nums[index + 1]:
                if p is not None:
                    return False
                p = i
        return (p is None or p == 0 or p == len(nums) - 2 or nums[p-1] <= A[P+1] or nums[p] <= nums[p+2])

File: 20190917/20190923/665. 非递减数列.cc
/*
 * Description: 非递减数列
 * Author: AskeyNil
 * CreateDate: 2019-09-23 15:24:06
 * LastEditors: AskeyNil
 *
 * *********************************
 * **                             **
 * **     　  你只有足够努力     　  **
 * **       才能看上去毫不费力       **
 * **                             **
 * *********************************
 *
 */

#include <iostream>
#include <vector>

using namespace std;

class Solution {
  public:
    bool checkPossibility(vector<int> &nums) {
        int p = -1;
        for (size_t i = 0; i < nums.size() - 1; i++) {
            if (nums[i] > nums[i + 1]) {
                if (p != -1)
                    return false;
                p = i;
            }
        }
        return (p == -1 || p == 0 || p == nums.size() - 2 ||
                nums[p - 1] <= nums[p + 1] || nums[p] <= nums[p + 2]);
    }
};